migrate to eslint v9

ryoppippi opened the pull request:
This PR includes updating eslint v9 flat config
This pull request includes significant changes to the ESLint configuration and several minor code updates to remove unnecessary ESLint disable comments.

### ESLint Configuration Updates:
* `eslint.config.cjs` was removed and replaced with `eslint.config.mjs`, which includes updated plugin imports, parser settings, and rule configurations. [[1]](diffhunk://#diff-ade92dc557e1c37f3e97d3323edfba82ec5ae154ff4325ddd06962631a5c2666L1-L28) [[2]](diffhunk://#diff-9601a8f6c734c2001be34a2361f76946d19a39a709b5e8c624a2a5a0aade05f2R1-R49)

### Package Configuration Updates:
* Updated `package.json` to include new ESLint dependencies and modify ESLint scripts to use caching for improved performance. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L24-R25) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R56-R57) [[3]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L66-R68)

### Code Cleanup:
* Removed unnecessary ESLint disable comments across various files to improve code cleanliness and maintainability. [[1]](diffhunk://#diff-e46dfbb9b3e32538422854508c462be9409bcf1b0a03e89aa205f965dc2a5aa2L76-R76) [[2]](diffhunk://#diff-3a8751f3d184886ba2003ca4fcd9dc6033a4288ef2c57f75916f96792fcc4f75R43) [[3]](diffhunk://#diff-bcaa0d2783a81ee5438c55b1d0fe08d75788c528c705b63d619644742dd5ab85R33) [[4]](diffhunk://#diff-06aa926702d76697ad98eb1eb573e64f45836bfc65d24a07e627a37620e24a2fL5-R5) [[5]](diffhunk://#diff-ab38b8e758e72d9310f44538be317da889d82719b978dcff06e2687ce508fd96L7-R7) [[6]](diffhunk://#diff-d3af1b00763cf212816ce058790def17b832381d03be82cbceee2d0224f219d6R92-R97) [[7]](diffhunk://#diff-c7d5e78a555b7e76c14766384b48c4ca898ed02e0606f2fbf3b62d421677ba66R266) [[8]](diffhunk://#diff-383f5967e48d40f45bc5e8b8efbfdfddf79035995aa010ddeedcf35b8308cef9R5) [[9]](diffhunk://#diff-804cc729e437660307ffe337f7dadce045a71583081079cc4ee8e0b5408ac725R181) [[10]](diffhunk://#diff-837877fe1d4f0ca65784db0ea2dff668d8d6f6377174c87fd4bb563609c0726bR72) [[11]](diffhunk://#diff-521b8d579f41292851b80c67f2a9a9668351137a284184761df4803abba2294fL2-R6) [[12]](diffhunk://#diff-6e0b70e733de6553c1ce9f69144ae2fe60355a47fccba2ff6a0dd3d68ebb09e7L10-R10) [[13]](diffhunk://#diff-6e0b70e733de6553c1ce9f69144ae2fe60355a47fccba2ff6a0dd3d68ebb09e7L33-R33)

ryoppippi committed:
feat(eslint): migrate to eslint v9

ryoppippi committed:
feat(eslint): only src

ryoppippi committed:
chore: eslint:fix

ryoppippi committed:
fix(eslint): use @typescript-eslint/no-deprecated instead of eslint-plugin-deprecation

https://typescript-eslint.io/rules/no-deprecated/
https://github.com/gund/eslint-plugin-deprecation

ryoppippi committed:
chore: ignore ts-deprecated

ryoppippi commented on src/factories/internal/metadata/iterate_metadata.ts:
```diff
@@ -40,6 +40,7 @@ export const iterate_metadata = (props: IMetadataIteratorProps): void => {
     return;
 
   // ITERATE CASES
+  // eslint-disable-next-line @typescript-eslint/no-unused-expressions
```
I disabled the lint error for now.
It's up to you if you want to delete it @samchon

ryoppippi commented on src/transformers/ImportTransformer.ts:
```diff
@@ -69,6 +69,7 @@ export namespace ImportTransformer {
       undefined,
       props.node.importClause,
       ts.factory.createStringLiteral(replaced),
+      // eslint-disable-next-line @typescript-eslint/no-deprecated
```
Got deprecated error

samchon commented on src/typings/ValueOf.ts:
The `boolean`, `number`, and `string` types are spec. This is invalid fixing.

ryoppippi replied:
OK, thanks!
I commited baeb63c16c5c

samchon commented on eslint.config.mjs:
I hope to enforce using only `const` keyword, and use `// disable-next-line` only when cannot escape using the `let` keyword.

samchon commented on eslint.config.mjs:
Also, the `eslint` must target every TypeScript files.

The only checking the `PROJECT_ROOT/src` directory was my mistake.

samchon reviewed (changes requested):
Thanks for contribution, but needs some revisions.

ryoppippi committed:
fix(eslint): disable `@typescript-eslint/no-wrapper-object-types`

https://github.com/samchon/typia/pull/1483/files/f14585f6a442c7d3ddccf4fcbd589b821fae2952#r1939732298

ryoppippi committed:
feat(eslint): gitignore

ryoppippi committed:
chore(eslint): format

ryoppippi committed:
feat(eslint): prefer-const

ryoppippi committed:
feat(eslint): semi

ryoppippi committed:
feat(eslint): no-var

ryoppippi committed:
feat(eslint): rewrite for flat config using typescript-eslint

https://typescript-eslint.io/getting-started/
https://typescript-eslint.io/getting-started/typed-linting/

ryoppippi committed:
chore(src): format

ryoppippi committed:
Merge branch 'master' into fix/eslint
